api/handlers_spaces.go

```diff
@@ -105,11 +106,25 @@ func (s *server) SpaceGetHandler(w http.ResponseWriter, r *http.Request) {
 		},
 	}
 
-	out, err := ceService.GetCostAndUsage(r.Context(), &input)
-	if err != nil {
-		msg := fmt.Sprintf("failed to get costs for space %s: %s", spaceID, err.Error())
-		handleError(w, errors.Wrap(err, msg))
-		return
+	// Use the go-cache if the object is found cached,
+	// else get the object in the cost-api delivering it to the cache
+	var out []*costexplorer.ResultByTime
+	c, ok := StatusCaches[account].Get(spaceID)
+	if !ok || c == nil {
+		log.Debugf("cache empty for Org: %s, calling Cost-API", Org)
+
+		// call cost-API
+		var err error
+		out, err = ceService.GetCostAndUsage(r.Context(), &input)
+		if err != nil {
+			msg := fmt.Sprintf("failed to get costs for space %s: %s", spaceID, err.Error())
+			handleError(w, errors.Wrap(err, msg))
+			return
+		}
+		StatusCaches[account].Set(spaceID, out, cache.DefaultExpiration)
+	} else {
+		out = c.([]*costexplorer.ResultByTime)
+		log.Debugf("found cached object: %s", out)
```

Comment:
this logic works and I think is what go-cache recommends... we could also cache the json `[]byte`.  we should keep an eye on memory utilization after we deploy to make sure everything is kosher.

Reply:
did nothing more so far